lib/bgs/exceptions/bgs_errors.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'bgs/exceptions/service_exception'
+module BGS
+  module Exceptions
+    module BGSErrors
+      include SentryLogging
+      MAX_ATTEMPTS = 3
+
+      def with_multiple_attempts_enabled
+        attempt ||= 0
+        yield
+      rescue => e
+        attempt += 1
+        if attempt < MAX_ATTEMPTS
+          notify_of_service_exception(e, __method__.to_s, attempt, :warn)
+          retry
+        end
+
+        notify_of_service_exception(e, __method__.to_s)
```

Comment:
You could move all the attempts logic into the method and simplify the `rescue` block. The nesting inside the `if` block and extra if with the return in the corresponding method might be indicating a refactoring opportunity.

Reply:
@johnpaulashenfelter Hey, sorry. I am not quite following you. I've made multiple attempts and can't figure out how to get all the attempts logic out of the rescue 🤔 . Could you elaborate a little more?